internal/pkg/githubapi/github.go

```diff
@@ -124,7 +124,14 @@ func HandlePREvent(eventPayload *github.PullRequestEvent, ghPrClientDetails GhPr
 					} else {
 						ghPrClientDetails.PrLogger.Debugf("PR %v labeled\n%+v", *eventPayload.PullRequest.Number, prLables)
 					}
-					// TODO Auto-merge PRs with no changes(optional)
+					if DoesPrHasLabel(*eventPayload, "promotion") && config.AutoMergeNoDiffPRs {
+						ghPrClientDetails.PrLogger.Infof("Auto-merging (no diff) PR %d", *eventPayload.PullRequest.Number)
+						err := MergePr(ghPrClientDetails, eventPayload.PullRequest.Number)
+						if err != nil {
+							prHandleError = err
```

Comment:
`errors.Join` == mind blown :)
I'll merge as is because I do log the errors so they are not completely lost but yeah, keeping them all is definitely better

Reply:
:0 I didn't even know about errors.Join

TechTime on error wrapping would definitely be killer. ErrorWrapping isn't something I've really picked up. I do love being able to use `errors.Is` for error pathing.